packages/circuit-ui/components/SelectorGroup/SelectorGroup.tsx

```diff
@@ -52,12 +57,100 @@ export interface SelectorGroupProps {
    * Whether the user can select multiple options.
    */
   multiple?: boolean;
+  /**
+   * Size of the Selectors within the group. Default: 'mega'.
+   */
+  size?: SelectorSize;
+  /**
+   * Spacing between selectors in a group. Default: 'kilo'.
+   */
+  gapSize?: GapSize;
+  /**
+   * Whether to display Selectors inline on mobile. Default: false.
+   */
+  inlineMobile?: boolean;
+  /**
+   * Whether the group should tale the whole width available when displayed in a row. Default: false.
+   */
+  stretch?: boolean;
   /**
    * The ref to the HTML Dom element
    */
   ref?: Ref<HTMLDivElement>;
 }
 
+type ContainerProps = Pick<
+  SelectorGroupProps,
+  'gapSize' | 'inlineMobile' | 'stretch'
+>;
+
+const stretchStyles = ({
+  theme,
+  stretch = false,
+}: StyleProps & ContainerProps) => css`
+  ${theme.mq.kilo} {
+    display: ${stretch ? 'flex' : 'inline-flex'};
+  }
+`;
+
+const inlineStyles = (theme: Theme, gapSize: GapSize) => css`
+  flex-direction: row;
+  width: auto;
+
+  > div {
+    &:not(:last-child) {
+      margin-right: ${theme.spacings[gapSize]};
+      margin-bottom: 0;
+    }
+  }
+`;
+
+const inlineMobileStyles = ({
+  theme,
+  gapSize = 'mega',
+  inlineMobile = false,
+}: StyleProps & ContainerProps) =>
+  inlineMobile &&
+  css`
+    label: selector-group--inline-mobile;
+    ${inlineStyles(theme, gapSize)}
+  `;
+
+const baseStyles = ({
+  theme,
+  gapSize = 'mega',
+}: StyleProps & ContainerProps) => css`
+  label: selector-group;
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+
+  > div {
+    &:not(:last-child) {
+      margin-bottom: ${theme.spacings[gapSize]};
+    }
+  }
+
+  ${theme.mq.kilo} {
+    ${inlineStyles(theme, gapSize)}
+  }
+`;
+
+const StyledSelectorGroup = styled.div(
+  baseStyles,
+  stretchStyles,
+  inlineMobileStyles,
+);
+
+const OptionItem = styled.div`
+  label: selector-group__option-item;
+  flex-grow: 1;
+`;
+
+const StyledSelector = styled(Selector)`
+  width: 100%;
+`;
```

Comment:
Why do we need this now? Because of the inline-block instead of block display property of the Selector? If we need to do this here, it means that someone who has implemented the `Selector` by itself in some settings might need a similar fix

Reply:
It's here to support the `stretch` prop. `Selector` needs to fill the width of `OptionItem` which has `flex-grow: 1;`, otherwise `OptionItem` will stretch but `Selector` within won't.